src/process.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2022 Igalia S.L.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#if defined(WPE_ENABLE_PROCESS) && WPE_ENABLE_PROCESS
+
+#include "../include/wpe/process.h"
+#include <stdlib.h>
+
+struct wpe_process_provider {
+    void* backend;
+};
+
+static struct wpe_process_provider_interface* provider_interface = NULL;
+
+struct wpe_process_provider*
+wpe_process_provider_create()
+{
+    if (!provider_interface)
+        return NULL;
+
+    struct wpe_process_provider* provider = calloc(1, sizeof(struct wpe_process_provider));
+    if (!provider)
+        return NULL;
+
+    if (provider_interface->create)
+        provider->backend = provider_interface->create(provider);
```

Comment:
@aperezdc i was following along with other WPE API conventions. Is there a better one in this case?

Reply:
The idea with this kind of API convention is to have a place where the provider can do additional initialization at the point when WebKit is instantiating the provider.

It is more useful for EGL initialization, and probably it will not be needed for process launching, but it's probably a good idea to have for consistency and in case it might be useful for some oddball system. Given that this is a singleton, having a couple more of pointers and a function call around doesn't cost much.